src/effects/native/reverbeffect.cpp

```diff
@@ -27,36 +27,61 @@ EffectManifest ReverbEffect::getManifest() {
     time->setId("bandwidth");
     time->setName(QObject::tr("Bandwidth"));
     time->setShortName(QObject::tr("BW"));
-    time->setDescription(QObject::tr("Higher bandwidth values cause more "
-            "bright (high-frequency) tones to be included"));
+    time->setDescription(QObject::tr("Bandwidth of the low pass filter at the input. "
+            "Higher values result in less attenuation of high frequencies."));
     time->setControlHint(EffectManifestParameter::CONTROL_KNOB_LINEAR);
     time->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
     time->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
-    time->setDefaultLinkType(EffectManifestParameter::LINK_LINKED);
-    time->setDefaultLinkInversion(EffectManifestParameter::LinkInversion::INVERTED);
-    time->setMinimum(0.0005);
-    time->setDefault(0.5);
-    time->setMaximum(1.0);
+    time->setMinimum(MIN_VALUE);
+    time->setDefault(MAX_VALUE);
+    time->setMaximum(MAX_VALUE);
+
+    EffectManifestParameter* decay = manifest.addParameter();
+    decay->setId("decay");
+    decay->setName(QObject::tr("Decay"));
+    decay->setDescription(QObject::tr("Lower decay values cause reverberations to die out more quickly."));
+    decay->setControlHint(EffectManifestParameter::CONTROL_KNOB_LINEAR);
+    decay->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    decay->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    decay->setMinimum(MIN_VALUE);
+    decay->setDefault(DEFAULT_DAMPING);
+    decay->setMaximum(MAX_VALUE);
 
     EffectManifestParameter* damping = manifest.addParameter();
     damping->setId("damping");
     damping->setName(QObject::tr("Damping"));
     damping->setDescription(QObject::tr("Higher damping values cause "
-            "reverberations to die out more quickly."));
+            "high frequencies to decay more quickly than low frequencies."));
     damping->setControlHint(EffectManifestParameter::CONTROL_KNOB_LINEAR);
     damping->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
     damping->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
-    damping->setMinimum(0.005);
-    damping->setDefault(0.5);
-    damping->setMaximum(1.0);
+    damping->setMinimum(MIN_VALUE);
+    damping->setDefault(MIN_VALUE);
+    damping->setMaximum(MAX_VALUE);
+
+    EffectManifestParameter* blend = manifest.addParameter();
```

Comment:
> Yes, if all effects needed the feature then sure it makes sense. But if only some need it (as in this case), then it makes more sense to define them in the effect.

I think there are other alternatives for the "this doesn't make sense in context" problem -- like having the effect indicate to the effect system whether it makes sense to show a D/W knob or send knob. And a per-effect D/W or send could be optionally presented in the GUI along-side the regular parameters when appropriate. 

> In this case the code is ported directly from the original "manufacturer", so it's been tested in production for more than 10 years, and we shouldn't have anything to worry about.

The implementation of blend here is guaranteed to not be vectorized due to the function call in the loop body (see the loop body disassembled below). The dry/wet here is also not ramping, so discontinuities can produce artifacts if the blend parameter is changed too quickly. There are benefits to providing the support in the effects system.

```
mixxx[0x100026db0] <+288>: leal   -0x1(%r13), %r12d
mixxx`MixxxPlateX2::processBuffer + 292 at Reverb.cc:453
   452 		for (uint i = 0; i + 1 < frames; i += 2) {
   453 			sample_t mono_sample = (in[i] + in[i + 1]) / 2;
   454 			sample_t xl, xr;
mixxx[0x100026db4] <+292>: movl   %r13d, %r14d
mixxx[0x100026db7] <+295>: movss  (%rax,%r14,4), %xmm0      ; xmm0 = mem[0],zero,zero,zero
mixxx`MixxxPlateX2::processBuffer + 301 at Reverb.cc:453
   452 		for (uint i = 0; i + 1 < frames; i += 2) {
   453 			sample_t mono_sample = (in[i] + in[i + 1]) / 2;
   454 			sample_t xl, xr;
mixxx[0x100026dbd] <+301>: addss  (%rax,%r12,4), %xmm0
mixxx`MixxxPlateX2::processBuffer + 307 at Reverb.cc:453
   452 		for (uint i = 0; i + 1 < frames; i += 2) {
   453 			sample_t mono_sample = (in[i] + in[i + 1]) / 2;
   454 			sample_t xl, xr;
mixxx[0x100026dc3] <+307>: mulss  0x68b4f5(%rip), %xmm0     ; typeinfo name for ReplayGain + 68
mixxx`MixxxPlateX2::processBuffer + 315 at Reverb.cc:455
   454 			sample_t xl, xr;
   455 			PlateStub::process(mono_sample, decay, &xl, &xr);
   456 			out[i] = blend*xl + dry*in[i];
mixxx[0x100026dcb] <+315>: movq   %r15, %rdi
mixxx[0x100026dce] <+318>: movss  -0x44(%rbp), %xmm1        ; xmm1 = mem[0],zero,zero,zero
mixxx[0x100026dd3] <+323>: movq   %r15, %rbx
mixxx[0x100026dd6] <+326>: movq   %rax, %r15
mixxx[0x100026dd9] <+329>: leaq   -0x2c(%rbp), %rsi
mixxx[0x100026ddd] <+333>: leaq   -0x30(%rbp), %rdx
mixxx[0x100026de1] <+337>: callq  0x100026500               ; PlateStub::process at Reverb.cc:245
mixxx[0x100026de6] <+342>: movq   %r15, %rax
mixxx[0x100026de9] <+345>: movq   %rbx, %r15
mixxx[0x100026dec] <+348>: movss  -0x2c(%rbp), %xmm0        ; xmm0 = mem[0],zero,zero,zero
mixxx[0x100026df1] <+353>: movss  -0x48(%rbp), %xmm3        ; xmm3 = mem[0],zero,zero,zero
mixxx`MixxxPlateX2::processBuffer + 358 at Reverb.cc:456
   455 			PlateStub::process(mono_sample, decay, &xl, &xr);
   456 			out[i] = blend*xl + dry*in[i];
   457 			out[i + 1] = blend*xr + dry*in[i + 1];
mixxx[0x100026df6] <+358>: mulss  %xmm3, %xmm0
mixxx[0x100026dfa] <+362>: movss  (%rax,%r12,4), %xmm1      ; xmm1 = mem[0],zero,zero,zero
mixxx[0x100026e00] <+368>: movss  -0x3c(%rbp), %xmm2        ; xmm2 = mem[0],zero,zero,zero
mixxx`MixxxPlateX2::processBuffer + 373 at Reverb.cc:456
   455 			PlateStub::process(mono_sample, decay, &xl, &xr);
   456 			out[i] = blend*xl + dry*in[i];
   457 			out[i + 1] = blend*xr + dry*in[i + 1];
mixxx[0x100026e05] <+373>: mulss  %xmm2, %xmm1
mixxx`MixxxPlateX2::processBuffer + 377 at Reverb.cc:456
   455 			PlateStub::process(mono_sample, decay, &xl, &xr);
   456 			out[i] = blend*xl + dry*in[i];
   457 			out[i + 1] = blend*xr + dry*in[i + 1];
mixxx[0x100026e09] <+377>: addss  %xmm0, %xmm1
mixxx`MixxxPlateX2::processBuffer + 381 at Reverb.cc:456
   455 			PlateStub::process(mono_sample, decay, &xl, &xr);
   456 			out[i] = blend*xl + dry*in[i];
   457 			out[i + 1] = blend*xr + dry*in[i + 1];
mixxx[0x100026e0d] <+381>: movq   -0x38(%rbp), %rcx
mixxx[0x100026e11] <+385>: movss  %xmm1, (%rcx,%r12,4)
mixxx[0x100026e17] <+391>: movq   -0x38(%rbp), %rcx
mixxx[0x100026e1b] <+395>: movss  -0x30(%rbp), %xmm0        ; xmm0 = mem[0],zero,zero,zero
mixxx`MixxxPlateX2::processBuffer + 400 at Reverb.cc:457
   456 			out[i] = blend*xl + dry*in[i];
   457 			out[i + 1] = blend*xr + dry*in[i + 1];
   458 		}
mixxx[0x100026e20] <+400>: mulss  %xmm3, %xmm0
mixxx[0x100026e24] <+404>: movss  (%rax,%r14,4), %xmm1      ; xmm1 = mem[0],zero,zero,zero
mixxx`MixxxPlateX2::processBuffer + 410 at Reverb.cc:457
   456 			out[i] = blend*xl + dry*in[i];
   457 			out[i + 1] = blend*xr + dry*in[i + 1];
   458 		}
mixxx[0x100026e2a] <+410>: mulss  %xmm2, %xmm1
mixxx`MixxxPlateX2::processBuffer + 414 at Reverb.cc:457
   456 			out[i] = blend*xl + dry*in[i];
   457 			out[i + 1] = blend*xr + dry*in[i + 1];
   458 		}
mixxx[0x100026e2e] <+414>: addss  %xmm0, %xmm1
mixxx`MixxxPlateX2::processBuffer + 418 at Reverb.cc:457
   456 			out[i] = blend*xl + dry*in[i];
   457 			out[i + 1] = blend*xr + dry*in[i + 1];
   458 		}
mixxx[0x100026e32] <+418>: movss  %xmm1, (%rcx,%r14,4)
mixxx[0x100026e38] <+424>: movl   -0x40(%rbp), %ecx
mixxx`MixxxPlateX2::processBuffer + 427 at Reverb.cc:452
   451 		// loop through the buffer, processing each sample
   452 		for (uint i = 0; i + 1 < frames; i += 2) {
   453 			sample_t mono_sample = (in[i] + in[i + 1]) / 2;
mixxx[0x100026e3b] <+427>: addl   $0x2, %r13d
mixxx[0x100026e3f] <+431>: cmpl   %ecx, %r13d
mixxx[0x100026e42] <+434>: jb     0x100026db0               ; <+288> at Reverb.cc:453
```